backend/inspirehep/orcid/domain_models.py

```diff
@@ -128,28 +128,21 @@ def push(self):
             # already exists (identified by the external identifiers).
             # This means we do not have the putcode, thus we cache all
             # author's putcodes and PUT the work again.
-            try:
-                if self.do_fail_if_duplicated_identifier:
-                    raise exceptions.DuplicatedExternalIdentifierPusherException
-                self._push_work_with_clashing_identifier()
-                putcode = self._post_or_put_work(putcode)
-            except orcid_client_exceptions.WorkAlreadyExistsException:
-                putcode = self._cache_all_author_putcodes()
-                if not putcode:
-                    msg = (
-                        "No putcode was found in ORCID API for orcid={} and recid={}."
-                        " And the POST has previously failed for the same recid because"
-                        " the work had already existed".format(self.orcid, self.recid)
-                    )
-                    raise exceptions.PutcodeNotFoundInOrcidException(msg)
-                putcode = self._post_or_put_work(putcode)
+            if self.pushing_duplicated_identifier:
+                raise exceptions.DuplicatedExternalIdentifierPusherException
```

Comment:
```suggestion
                try:
                    putcode = self._cache_putcodes_and_put_work()
                except orcid_client_exceptions.WorkAlreadyExistsException:
                     raise exceptions.DuplicatedExternalIdentifierPusherException
```
as you might be missing the putcode when pushing the clashing identifier (you should add a test for this!)